prov/cxi/src/cxip_av.c

```diff
@@ -85,10 +85,9 @@ static int cxip_av_insert_addr(struct cxip_av *av, struct cxip_addr *addr,
 {
 	struct cxip_av_entry *entry;
 	struct cxip_av_auth_key_entry *auth_key_entry = NULL;
-	struct cxip_addr auth_key_addr = {
-		.nic = addr->nic,
-		.pid = addr->pid
-	};
+	struct cxip_addr auth_key_addr = {0};
```

Comment:
Hi Thomas, thanks for finding this. Would you consider changing structure cxip_addr instead of the proposed fix:

struct cxip_addr {
        uint32_t pid            : C_DFA_PID_BITS_MAX;
        uint32_t nic            : C_DFA_NIC_BITS;
        uint32_t pad.          : 3;
        uint16_t vni;
};

and see if this fixes the initialization issue. Also we use stack structure initialization of {} (C23) instead of {0} and would like to maintain that usage. Thanks.

Reply:
@swelch, while this might work, my point of view is with regards to the clarity of the code. explicitly initializing the local variable to zero is more clear. It's your code at the end... but my preference would be for an explicit initialization in the code. This will also continue to work in the unlikely event that the cxip_addr structure itself changes.

Reply:
just to be clear. I think adding the padding is good. I'm just concerned about the clarity of the code. IE any structure used in the hashing needs to be zeroed out properly.

Reply:
@amirshehataornl  I would just like the bug to be addressed by adding the pad to the struct. That way either method works. As far as how you want to initialize I don't have a strong opinion how you initialize as both ways would then work. Thanks again.